src/components/ProductCard/components/CloseButton.jsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useAuth } from "../../../context/auth-context";
+import { useWishlist } from "../../../context/wishlist-context";
+
+export const CloseButton = ({ product }) => {
+  const { authData } = useAuth();
+  const [isDisabled, setDisabled] = useState(false);
+  const { handleDeleteWishlist } = useWishlist();
+  const navigate = useNavigate();
+
+  return (
+    <button
+      className="btn icon--btn icon--badge"
+      disabled={isDisabled}
+      onClick={() => {
+        if (!authData.isLoggedIn) {
+          navigate("/login");
+        }
+        if (authData.isLoggedIn) {
+          setDisabled((prev) => !prev);
+          handleDeleteWishlist(product);
```

Comment:
use ternary instead of `if`

Reply:
Thanks @samflab for detailed review made most of the change